Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -856,6 +856,10 @@ void WebProcessProxy::addExistingWebPage(WebPageProxy& webPage, BeginsUsingDataS
     updateRegistrationWithDataStore();
     updateBackgroundResponsivenessTimer();
     protectedWebsiteDataStore()->propagateSettingUpdatesToNetworkProcess();
+#if HAVE(ALLOW_ONLY_PARTITIONED_COOKIES)
+    if (auto [didChange, enabled] = protectedWebsiteDataStore()->didOptInCookiePartitioningStateChange(); didChange)
+        protectedProcessPool()->sendToAllProcesses(Messages::WebProcess::SetOptInCookiePartitioningEnabled(enabled));
```

Comment:
Web content process needs to get `WebsiteDataStore::m_isOptInCookiePartitioningEnabled` value on initialization (likely in `WebProcess::SetWebsiteDataStoreParameters`), and UI process will send out message on update.

Also `WebsiteDataStore:: m_isOptInCookiePartitioningEnabled ` can be updated in 3 cases:
1. page is added
2. page is removed
3. page's `isOptInCookiePartitioningEnabled` preference value has changed
It seems we don't update the value in the third case.